src/languages/cpp.js

```diff
@@ -429,7 +480,28 @@ export default function(hljs) {
       /(?!switch)/,
       /(?!while)/,
       hljs.IDENT_RE,
-      regex.lookahead(/(<[^<>]+>|)\s*\(/))
+      regex.lookahead(/(<[^<>]+>|)\s*\(/)),
+    returnBegin: 1,
+    contains:[
```

Comment:
I see now you're rewinding then parsing again, but why did you find this necessary?